# integrantes: Jesús David Angulo Reyes
 Carlos Andres Mendoza Martinez


# primer examen
Primero, investigue y analice cómo realizar un análisis de marca. Luego desarrolle un análisis sobre la tendencia de la moda de productos de ropa interior y trajes de baño. Los conjuntos de datos se crearon mediante la extracción de datos de sitios minoristas populares a través de las soluciones de extracción de datos de PromptCloud. Los sitios cubiertos son; Amazon, Victoria's Secret, Btemptd, Calvin Klein,  Hanky ​​Panky, American Eagle, Macy's, Nordstrom, Topshop USA. Además, algunas de las preguntas más comunes que se pueden responder son: 


* ¿Cómo difiere el precio según la marca?
* Modelado de temas en la descripción del producto.
* ¿Cuáles son los colores más comunes utilizados por las diferentes marcas?
* Análisis de las calificaciones de los productos (cuando corresponda)
* Atributos de estilo comunes (cuando corresponda)
* Finalmente, realizar un análisis predictivo teniendo en cuenta los algoritmos vistos en clase.

In [26]:
# librerias usadas
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn import linear_model

# cargar los dataset existentes

In [81]:
#Leemos el archivo csv y lo asignamos mediante Pandas a la variable dataframe
#Mediante el método dataframe.head() vemos en pantalla los 5 primeros registros

data_victoriassecret = pd.read_csv('victoriassecret_com.csv', index_col=0)
data_us_topshop_com = pd.read_csv('us_topshop_com.csv', index_col=0)
data_shop_nordstrom_com = pd.read_csv('shop_nordstrom_com.csv', index_col=0)
data_macys_com = pd.read_csv('macys_com.csv', index_col=0)
data_hankypanky_com = pd.read_csv('hankypanky_com.csv', index_col=0)
data_calvinklein_com = pd.read_csv('calvinklein_com.csv', index_col=0)
data_btemptd_com = pd.read_csv('btemptd_com.csv', index_col=0)
data_amazon_com = pd.read_csv('amazon_com.csv', index_col=0)
data_ae_com = pd.read_csv('ae_com.csv', index_col=0)

In [82]:
#Victorias Secret
data_victoriassecret.head()

mrp    price  \
product_name                                           
Very Sexy Strappy Lace Thong Panty  $14.50   $14.50    
Very Sexy Strappy Lace Thong Panty  $14.50   $14.50    
Very Sexy Strappy Lace Thong Panty  $14.50   $14.50    
Very Sexy Strappy Lace Thong Panty  $14.50   $14.50    
Very Sexy Strappy Lace Thong Panty  $14.50   $14.50    

                                                                              pdp_url  \
product_name                                                                            
Very Sexy Strappy Lace Thong Panty  https://www.victoriassecret.com/panties/shop-a...   
Very Sexy Strappy Lace Thong Panty  https://www.victoriassecret.com/panties/shop-a...   
Very Sexy Strappy Lace Thong Panty  https://www.victoriassecret.com/panties/shop-a...   
Very Sexy Strappy Lace Thong Panty  https://www.victoriassecret.com/panties/shop-a...   
Very Sexy Strappy Lace Thong Panty  https://www.victoriassecret.com/panties/shop-a...   

                                           brand_name  \
product_name                                            
Very Sexy Strappy Lace Thong Panty  Victoria's Secret   
Very Sexy Strappy Lace Thong Panty  Victoria's Secret   
Very Sexy Strappy Lace Thong Panty  Victoria's Secret   
Very Sexy Strappy Lace Thong Panty  Victoria's Secret   
Very Sexy Strappy Lace Thong Panty  Victoria's Secret   

                                            product_category  \
product_name                                                   
Very Sexy Strappy Lace Thong Panty  Strappy Lace Thong Panty   
Very Sexy Strappy Lace Thong Panty  Strappy Lace Thong Panty   
Very Sexy Strappy Lace Thong Panty  Strappy Lace Thong Panty   
Very Sexy Strappy Lace Thong Panty  Strappy Lace Thong Panty   
Very Sexy Strappy Lace Thong Panty  Strappy Lace Thong Panty   

                                              retailer  \
product_name                                             
Very Sexy Strappy Lace Thong Panty  Victoriassecret US   
Very Sexy Strappy Lace Thong Panty  Victoriassecret US   
Very Sexy Strappy Lace Thong Panty  Victoriassecret US   
Very Sexy Strappy Lace Thong Panty  Victoriassecret US   
Very Sexy Strappy Lace Thong Panty  Victoriassecret US   

                                                                          description  \
product_name                                                                            
Very Sexy Strappy Lace Thong Panty  Lots of cheek peek, pretty lace, a strappy bac...   
Very Sexy Strappy Lace Thong Panty  Lots of cheek peek, pretty lace, a strappy bac...   
Very Sexy Strappy Lace Thong Panty  Lots of cheek peek, pretty lace, a strappy bac...   
Very Sexy Strappy Lace Thong Panty  Lots of cheek peek, pretty lace, a strappy bac...   
Very Sexy Strappy Lace Thong Panty  Lots of cheek peek, pretty lace, a strappy bac...   

                                    rating  review_count  style_attributes  \
product_name                                                                 
Very Sexy Strappy Lace Thong Panty     NaN           NaN               NaN   
Very Sexy Strappy Lace Thong Panty     NaN           NaN               NaN   
Very Sexy Strappy Lace Thong Panty     NaN           NaN               NaN   
Very Sexy Strappy Lace Thong Panty     NaN           NaN               NaN   
Very Sexy Strappy Lace Thong Panty     NaN           NaN               NaN   

                                                    total_sizes  \
product_name                                                      
Very Sexy Strappy Lace Thong Panty  ["XS", "S", "M", "L", "XL"]   
Very Sexy Strappy Lace Thong Panty  ["XS", "S", "M", "L", "XL"]   
Very Sexy Strappy Lace Thong Panty  ["XS", "S", "M", "L", "XL"]   
Very Sexy Strappy Lace Thong Panty  ["XS", "S", "M", "L", "XL"]   
Very Sexy Strappy Lace Thong Panty  ["XS", "S", "M", "L", "XL"]   

                                   available_size        color  
product_name                                                 

In [8]:
data_us_topshop_com.head()

mrp    price  \
product_name                                       
MATERNITY Criss Cross Knickers   $20.00   $20.00   
Floral Lace Triangle Bra         $35.00   $35.00   
Floral Lace Thong                $20.00   $20.00   
Mesh French Knickers             $10.00   $10.00   
Floral Lace Thong                $10.00   $10.00   

                                                                          pdp_url  \
product_name                                                                        
MATERNITY Criss Cross Knickers  http://www.topshop.com/en/tsus/product/clothin...   
Floral Lace Triangle Bra        http://www.topshop.com/en/tsus/product/clothin...   
Floral Lace Thong               http://www.topshop.com/en/tsus/product/clothin...   
Mesh French Knickers            http://www.topshop.com/en/tsus/product/clothin...   
Floral Lace Thong               http://www.topshop.com/en/tsus/product/clothin...   

                                brand_name product_category    retailer  \
product_name                                                              
MATERNITY Criss Cross Knickers  US TOPSHOP         Lingerie  Topshop US   
Floral Lace Triangle Bra        US TOPSHOP         Lingerie  Topshop US   
Floral Lace Thong               US TOPSHOP         Lingerie  Topshop US   
Mesh French Knickers            US TOPSHOP         Lingerie  Topshop US   
Floral Lace Thong               US TOPSHOP         Lingerie  Topshop US   

                                                                      description  \
product_name                                                                        
MATERNITY Criss Cross Knickers  These feminine black knickers for maternity fe...   
Floral Lace Triangle Bra        In an appealing burnt orange tint, this low ri...   
Floral Lace Thong               In an appealing burnt orange tint, this low ri...   
Mesh French Knickers            Indulge your playful side with these sheer mes...   
Floral Lace Thong               Keep slinky silhouettes seam-free with this la...   

                                rating  review_count  style_attributes  \
product_name                                                             
MATERNITY Criss Cross Knickers     NaN           NaN               NaN   
Floral Lace Triangle Bra           NaN           NaN               NaN   
Floral Lace Thong                  NaN           NaN               NaN   
Mesh French Knickers               NaN           NaN               NaN   
Floral Lace Thong                  3.0           1.0               NaN   

                                     total_sizes available_size      color  
product_name                                                                
MATERNITY Criss Cross Knickers       4,6,8,10,12    4,6,8,10,12      BLACK  
Floral Lace Triangle Bra           2,4,6,8,10,12      2,8,10,12        NaN  
Floral Lace Thong                  2,4,6,8,10,12  2,4,6,8,10,12        NaN  
Mesh French Knickers            2,4,6,8,10,12,14      4,6,10,12  RASPBERRY  
Floral Lace Thong                  2,4,6,8,10,12  2,4,6,8,10,12      BLACK

In [9]:
data_shop_nordstrom_com.head()

mrp    price  \
product_name                                             
'B Fitting' High Cut Briefs           $15.00   $15.00    
'B Fitting' High Cut Briefs           $15.00   $15.00    
'B Fitting' High Cut Briefs           $15.00   $15.00    
'B Fitting' High Cut Briefs           $15.00   $15.00    
'Skinsense' Seamless High Cut Briefs  $18.00   $18.00    

                                                                                pdp_url  \
product_name                                                                              
'B Fitting' High Cut Briefs           http://shop.nordstrom.com/s/wacoal-b-fitting-h...   
'B Fitting' High Cut Briefs           http://shop.nordstrom.com/s/wacoal-b-fitting-h...   
'B Fitting' High Cut Briefs           http://shop.nordstrom.com/s/wacoal-b-fitting-h...   
'B Fitting' High Cut Briefs           http://shop.nordstrom.com/s/wacoal-b-fitting-h...   
'Skinsense' Seamless High Cut Briefs  http://shop.nordstrom.com/s/wacoal-skinsense-s...   

                                     brand_name    product_category  \
product_name                                                          
'B Fitting' High Cut Briefs              WACOAL     Women's Panties   
'B Fitting' High Cut Briefs              WACOAL     Women's Panties   
'B Fitting' High Cut Briefs              WACOAL     Women's Panties   
'B Fitting' High Cut Briefs              WACOAL     Women's Panties   
'Skinsense' Seamless High Cut Briefs     WACOAL  Skin-Tone Lingerie   

                                          retailer  \
product_name                                         
'B Fitting' High Cut Briefs           Nordstrom US   
'B Fitting' High Cut Briefs           Nordstrom US   
'B Fitting' High Cut Briefs           Nordstrom US   
'B Fitting' High Cut Briefs           Nordstrom US   
'Skinsense' Seamless High Cut Briefs  Nordstrom US   

                                                                            description  \
product_name                                                                              
'B Fitting' High Cut Briefs           Lighter-than-air, full-cut Supima® cotton brie...   
'B Fitting' High Cut Briefs           Lighter-than-air, full-cut Supima® cotton brie...   
'B Fitting' High Cut Briefs           Lighter-than-air, full-cut Supima® cotton brie...   
'B Fitting' High Cut Briefs           Lighter-than-air, full-cut Supima® cotton brie...   
'Skinsense' Seamless High Cut Briefs  Cleanly finished edges ensure that these full-...   

                                      rating  review_count  \
product_name                                                 
'B Fitting' High Cut Briefs              4.2          65.0   
'B Fitting' High Cut Briefs              4.2          65.0   
'B Fitting' High Cut Briefs              4.2          65.0   
'B Fitting' High Cut Briefs              4.2          65.0   
'Skinsense' Seamless High Cut Briefs     3.0           2.0   

                                                                       style_attributes  \
product_name                                                                              
'B Fitting' High Cut Briefs           ["79% Supima® cotton, 21% spandex.", "Hand was...   
'B Fitting' High Cut Briefs           ["79% Supima® cotton, 21% spandex.", "Hand was...   
'B Fitting' High Cut Briefs           ["79% Supima® cotton, 21% spandex.", "Hand was...   
'B Fitting' High Cut Briefs           ["79% Supima® cotton, 21% spandex.", "Hand was...   
'Skinsense' Seamless High Cut Briefs  ["Cotton-lined gusset.", "72% nylon, 28% spand...   

                                                                  total_sizes  \
product_name                                                                    
'B Fitting' High Cut Briefs                                             [nil]   
'B Fitting' High Cut Briefs                                             [nil]   
'B Fitting' High Cut Briefs                                             [nil]   
'B F

In [10]:
data_macys_com.head()

mrp    price  \
product_name                                                           
ID String Bikini QF1754                             $20.00   $20.00    
ID String Bikini QF1754                             $20.00   $20.00    
ID String Bikini QF1754                             $20.00   $20.00    
CK Black Collection Embrace Lace-Waist Thong QF...  $26.00   $26.00    
Halo Lace Boyshort 870205                           $15.00   $15.00    

                                                                                              pdp_url  \
product_name                                                                                            
ID String Bikini QF1754                             http://www1.macys.com/shop/product/calvin-klei...   
ID String Bikini QF1754                             http://www1.macys.com/shop/product/calvin-klei...   
ID String Bikini QF1754                             http://www1.macys.com/shop/product/calvin-klei...   
CK Black Collection Embrace Lace-Waist Thong QF...  http://www1.macys.com/shop/product/calvin-klei...   
Halo Lace Boyshort 870205                           http://www1.macys.com/shop/product/wacoal-halo...   

                                                      brand_name  \
product_name                                                       
ID String Bikini QF1754                             Calvin Klein   
ID String Bikini QF1754                             Calvin Klein   
ID String Bikini QF1754                             Calvin Klein   
CK Black Collection Embrace Lace-Waist Thong QF...  Calvin Klein   
Halo Lace Boyshort 870205                                 Wacoal   

                                                                                    product_category  \
product_name                                                                                           
ID String Bikini QF1754                             Women - Lingerie & Shapewear - Designer Lingerie   
ID String Bikini QF1754                             Women - Lingerie & Shapewear - Designer Lingerie   
ID String Bikini QF1754                             Women - Lingerie & Shapewear - Designer Lingerie   
CK Black Collection Embrace Lace-Waist Thong QF...  Women - Lingerie & Shapewear - Designer Lingerie   
Halo Lace Boyshort 870205                           Women - Lingerie & Shapewear - Designer Lingerie   

                                                    retailer  \
product_name                                                   
ID String Bikini QF1754                             Macys US   
ID String Bikini QF1754                             Macys US   
ID String Bikini QF1754                             Macys US   
CK Black Collection Embrace Lace-Waist Thong QF...  Macys US   
Halo Lace Boyshort 870205                           Macys US   

                                                                                          description  \
product_name                                                                                            
ID String Bikini QF1754                             The perfect amount of coverage in a subtle sil...   
ID String Bikini QF1754                             The perfect amount of coverage in a subtle sil...   
ID String Bikini QF1754                             The perfect amount of coverage in a subtle sil...   
CK Black Collection Embrace Lace-Waist Thong QF...  Exquisitely designed embroidered lace beautifu...   
Halo Lace Boyshort 870205                           A fit that is heavenly. The stretch lace Halo ...   

                                                    rating  review_count  \
product_name                                                               
ID String Bikini QF1754                                NaN           NaN   
ID String Bikini QF1754                                NaN           NaN   
ID String Bikini QF1754                                NaN           NaN   
CK Black Collection Embrace Lace-Waist Thong

In [11]:
data_hankypanky_com.head()

mrp price  \
product_name                                       
Silky 20” A-line Half Slip with Lace  $68   $68    
Silky 20” A-line Half Slip with Lace  $68   $68    
Peek-a-Boo Side Tie Bikini            $30   $30    
Peek-a-Boo Side Tie Bikini            $30   $30    
All Tied Up Tie Front Bralette        $59   $59    

                                                                                pdp_url  \
product_name                                                                              
Silky 20” A-line Half Slip with Lace  http://www.hankypanky.com/collections/silky-20...   
Silky 20” A-line Half Slip with Lace  http://www.hankypanky.com/collections/silky-20...   
Peek-a-Boo Side Tie Bikini            http://www.hankypanky.com/panties/bridal-peek-...   
Peek-a-Boo Side Tie Bikini            http://www.hankypanky.com/panties/bridal-peek-...   
All Tied Up Tie Front Bralette        http://www.hankypanky.com/collections/all-tied...   

                                      brand_name product_category  \
product_name                                                        
Silky 20” A-line Half Slip with Lace  HankyPanky      Collections   
Silky 20” A-line Half Slip with Lace  HankyPanky      Collections   
Peek-a-Boo Side Tie Bikini            HankyPanky          Panties   
Peek-a-Boo Side Tie Bikini            HankyPanky          Panties   
All Tied Up Tie Front Bralette        HankyPanky      Collections   

                                           retailer  \
product_name                                          
Silky 20” A-line Half Slip with Lace  Hankypanky US   
Silky 20” A-line Half Slip with Lace  Hankypanky US   
Peek-a-Boo Side Tie Bikini            Hankypanky US   
Peek-a-Boo Side Tie Bikini            Hankypanky US   
All Tied Up Tie Front Bralette        Hankypanky US   

                                                                            description  \
product_name                                                                              
Silky 20” A-line Half Slip with Lace  Hanky Panky Silky is the ideal fabric for unde...   
Silky 20” A-line Half Slip with Lace  Hanky Panky Silky is the ideal fabric for unde...   
Peek-a-Boo Side Tie Bikini            Don't go on your honeymoon without this pretty...   
Peek-a-Boo Side Tie Bikini            Don't go on your honeymoon without this pretty...   
All Tied Up Tie Front Bralette                         All Tied Up in ribbons and lace.   

                                      rating  review_count  \
product_name                                                 
Silky 20” A-line Half Slip with Lace     NaN           NaN   
Silky 20” A-line Half Slip with Lace     NaN           NaN   
Peek-a-Boo Side Tie Bikini               NaN           NaN   
Peek-a-Boo Side Tie Bikini               NaN           NaN   
All Tied Up Tie Front Bralette           NaN           NaN   

                                                                       style_attributes  \
product_name                                                                              
Silky 20” A-line Half Slip with Lace  ["Just above-the-knee-length half-slip with el...   
Silky 20” A-line Half Slip with Lace  ["Just above-the-knee-length half-slip with el...   
Peek-a-Boo Side Tie Bikini            ["Bikini-style panty in sheer lace ", " Featur...   
Peek-a-Boo Side Tie Bikini            ["Bikini-style panty in sheer lace ", " Featur...   
All Tied Up Tie Front Bralette        ["Bralette in scalloped cross-dyed lace", " Fl...   

                                                          total_sizes  \
product_name                                                            
Silky 20” A-line Half Slip with Lace  ["Select", "S", "M", "L", "XL"]   
Silky 20” A-line Half Slip with Lace  ["Select", "S", "M", "L", "XL"]   
Peek-a-Boo Side Tie Bikini                  ["Select", "S", "M", "L"]   
Peek-a-Boo Side Tie Bikini                  ["Select", "S", "M", "L"]   
All Tied Up Tie F

In [12]:
data_calvinklein_com.head()

mrp    price  \
product_name                                  
logo cotton stretch thong  $13.00   $13.00    
logo cotton stretch thong  $13.00   $13.00    
logo cotton stretch thong  $13.00   $13.00    
logo cotton stretch thong  $13.00   $13.00    
essentials thong           $13.00   $13.00    

                                                                     pdp_url  \
product_name                                                                   
logo cotton stretch thong  http://www.calvinklein.us/en/womens-clothing/w...   
logo cotton stretch thong  http://www.calvinklein.us/en/womens-clothing/w...   
logo cotton stretch thong  http://www.calvinklein.us/en/womens-clothing/w...   
logo cotton stretch thong  http://www.calvinklein.us/en/womens-clothing/w...   
essentials thong           http://www.calvinklein.us/en/womens-clothing/w...   

                             brand_name           product_category  \
product_name                                                         
logo cotton stretch thong  Calvin Klein  3 FOR 33 PANTY ESSENTIALS   
logo cotton stretch thong  Calvin Klein  3 FOR 33 PANTY ESSENTIALS   
logo cotton stretch thong  Calvin Klein  3 FOR 33 PANTY ESSENTIALS   
logo cotton stretch thong  Calvin Klein  3 FOR 33 PANTY ESSENTIALS   
essentials thong           Calvin Klein                    Panties   

                                  retailer  \
product_name                                 
logo cotton stretch thong  Calvin Klein US   
logo cotton stretch thong  Calvin Klein US   
logo cotton stretch thong  Calvin Klein US   
logo cotton stretch thong  Calvin Klein US   
essentials thong           Calvin Klein US   

                                                                 description  \
product_name                                                                   
logo cotton stretch thong  soft cotton stretch fabric and a metallic logo...   
logo cotton stretch thong  soft cotton stretch fabric and a metallic logo...   
logo cotton stretch thong  soft cotton stretch fabric and a metallic logo...   
logo cotton stretch thong  soft cotton stretch fabric and a metallic logo...   
essentials thong           the essentials thong features a semi sheer pat...   

                           rating  review_count  \
product_name                                      
logo cotton stretch thong     NaN           NaN   
logo cotton stretch thong     NaN           NaN   
logo cotton stretch thong     NaN           NaN   
logo cotton stretch thong     NaN           NaN   
essentials thong              NaN           NaN   

                                                            style_attributes  \
product_name                                                                   
logo cotton stretch thong  ["cotton stretch thong panty","metallic elasti...   
logo cotton stretch thong  ["cotton stretch thong panty","metallic elasti...   
logo cotton stretch thong  ["cotton stretch thong panty","metallic elasti...   
logo cotton stretch thong  ["cotton stretch thong panty","metallic elasti...   
essentials thong           ["semi sheer thong panty","thin elastic logo w...   

                          total_sizes available_size  \
product_name                                           
logo cotton stretch thong       s,m,l          s,m,l   
logo cotton stretch thong       s,m,l          s,m,l   
logo cotton stretch thong       s,m,l          s,m,l   
logo cotton stretch thong       s,m,l          s,m,l   
essentials thong                s,m,l          s,m,l   

                                                      color  
product_name                                                 
logo cotton stretch thong                             BLACK  
logo cotton stretch thong                      ASHFORD GREY  
logo cotton stretch thong                             WHITE  
logo cotton stretch thong                              NAVY  
essentials thong           EPHEMERAL WITH IVORY LACE & LOGO

In [13]:
data_btemptd_com.head()

mrp    price  \
product_name                                      
b.tempt'd Ciao Bella Bralette  $30.00   $30.00    
b.tempt'd Ciao Bella Bralette  $30.00   $30.00    
b.tempt'd Ciao Bella Bralette  $30.00   $30.00    
b.tempt'd Ciao Bella Bralette  $30.00   $21.00    
b.tempt'd Ciao Bella Bralette  $30.00   $21.00    

                                                                         pdp_url  \
product_name                                                                       
b.tempt'd Ciao Bella Bralette  http://btemptd.wacoal-america.com/b-tempt-d-ci...   
b.tempt'd Ciao Bella Bralette  http://btemptd.wacoal-america.com/b-tempt-d-ci...   
b.tempt'd Ciao Bella Bralette  http://btemptd.wacoal-america.com/b-tempt-d-ci...   
b.tempt'd Ciao Bella Bralette  http://btemptd.wacoal-america.com/b-tempt-d-ci...   
b.tempt'd Ciao Bella Bralette  http://btemptd.wacoal-america.com/b-tempt-d-ci...   

                              brand_name product_category    retailer  \
product_name                                                            
b.tempt'd Ciao Bella Bralette     WACOAL      COLLECTIONS  Btemptd US   
b.tempt'd Ciao Bella Bralette     WACOAL      COLLECTIONS  Btemptd US   
b.tempt'd Ciao Bella Bralette     WACOAL      COLLECTIONS  Btemptd US   
b.tempt'd Ciao Bella Bralette     WACOAL      COLLECTIONS  Btemptd US   
b.tempt'd Ciao Bella Bralette     WACOAL      COLLECTIONS  Btemptd US   

                                                                     description  \
product_name                                                                       
b.tempt'd Ciao Bella Bralette  Say “buongiorno!” to this ladylike piece that ...   
b.tempt'd Ciao Bella Bralette  Say “buongiorno!” to this ladylike piece that ...   
b.tempt'd Ciao Bella Bralette  Say “buongiorno!” to this ladylike piece that ...   
b.tempt'd Ciao Bella Bralette  Say “buongiorno!” to this ladylike piece that ...   
b.tempt'd Ciao Bella Bralette  Say “buongiorno!” to this ladylike piece that ...   

                               rating  review_count  \
product_name                                          
b.tempt'd Ciao Bella Bralette     NaN           NaN   
b.tempt'd Ciao Bella Bralette     NaN           NaN   
b.tempt'd Ciao Bella Bralette     NaN           NaN   
b.tempt'd Ciao Bella Bralette     NaN           NaN   
b.tempt'd Ciao Bella Bralette     NaN           NaN   

                                                                style_attributes  \
product_name                                                                       
b.tempt'd Ciao Bella Bralette  [Wire free bralette • Cut and sew corded lace ...   
b.tempt'd Ciao Bella Bralette  [Wire free bralette • Cut and sew corded lace ...   
b.tempt'd Ciao Bella Bralette  [Wire free bralette • Cut and sew corded lace ...   
b.tempt'd Ciao Bella Bralette  [Wire free bralette • Cut and sew corded lace ...   
b.tempt'd Ciao Bella Bralette  [Wire free bralette • Cut and sew corded lace ...   

                               total_sizes available_size          color  
product_name                                                              
b.tempt'd Ciao Bella Bralette  XS,S,M,L,XL    l,m,s,xl,xs   Bridal White  
b.tempt'd Ciao Bella Bralette  XS,S,M,L,XL    l,m,s,xl,xs          Night  
b.tempt'd Ciao Bella Bralette  XS,S,M,L,XL    l,m,s,xl,xs     Wild Aster  
b.tempt'd Ciao Bella Bralette  XS,S,M,L,XL    l,m,s,xl,xs  Camellia Rose  
b.tempt'd Ciao Bella Bralette  XS,S,M,L,XL    l,m,s,xl,xs     Cappuccino

In [14]:
data_amazon_com.head()

mrp    price  \
product_name                                                           
Calvin Klein Women's Sheer Marquisette Demi Unl...  $36.00   $32.40    
Wacoal Embrace Lace Bikini Panty                    $27.00   $27.00    
Wacoal Women's Slimline Seamless Minimizer Bra      $65.00   $65.00    
Hanky Panky Womens Signature Lace Retro V-Kini      $36.00   $36.00    
Wacoal Women's Red Carpet Strapless Bra             $65.00   $65.00    

                                                                                              pdp_url  \
product_name                                                                                            
Calvin Klein Women's Sheer Marquisette Demi Unl...  https://www.amazon.com/-/dp/B01NAVD98J?th=1&psc=1   
Wacoal Embrace Lace Bikini Panty                    https://www.amazon.com/-/dp/B0011YQFNK?th=1&psc=1   
Wacoal Women's Slimline Seamless Minimizer Bra      https://www.amazon.com/-/dp/B000T3606Q?th=1&psc=1   
Hanky Panky Womens Signature Lace Retro V-Kini      https://www.amazon.com/-/dp/B003Y6AX0Y?th=1&psc=1   
Wacoal Women's Red Carpet Strapless Bra             https://www.amazon.com/-/dp/B01CEBGQA0?th=1&psc=1   

                                                      brand_name  \
product_name                                                       
Calvin Klein Women's Sheer Marquisette Demi Unl...  Calvin-Klein   
Wacoal Embrace Lace Bikini Panty                          Wacoal   
Wacoal Women's Slimline Seamless Minimizer Bra            Wacoal   
Hanky Panky Womens Signature Lace Retro V-Kini       Hanky-Panky   
Wacoal Women's Red Carpet Strapless Bra                   Wacoal   

                                                   product_category  \
product_name                                                          
Calvin Klein Women's Sheer Marquisette Demi Unl...             Bras   
Wacoal Embrace Lace Bikini Panty                            Panties   
Wacoal Women's Slimline Seamless Minimizer Bra                 Bras   
Hanky Panky Womens Signature Lace Retro V-Kini              Panties   
Wacoal Women's Red Carpet Strapless Bra                        Bras   

                                                     retailer  \
product_name                                                    
Calvin Klein Women's Sheer Marquisette Demi Unl...  Amazon US   
Wacoal Embrace Lace Bikini Panty                    Amazon US   
Wacoal Women's Slimline Seamless Minimizer Bra      Amazon US   
Hanky Panky Womens Signature Lace Retro V-Kini      Amazon US   
Wacoal Women's Red Carpet Strapless Bra             Amazon US   

                                                                                          description  \
product_name                                                                                            
Calvin Klein Women's Sheer Marquisette Demi Unl...  An unlined demi cup bra featuring sheer, sexy ...   
Wacoal Embrace Lace Bikini Panty                    Embrace lace bikini offers great fit and match...   
Wacoal Women's Slimline Seamless Minimizer Bra      Seamless underwire minimizer bra gives great s...   
Hanky Panky Womens Signature Lace Retro V-Kini      All-day comfort describes this figure-flatteri...   
Wacoal Women's Red Carpet Strapless Bra             Red Carpet full figure strapless fits great, s...   

                                                    rating  review_count  \
product_name                                                               
Calvin Klein Women's Sheer Marquisette Demi Unl...     4.5            47   
Wacoal Embrace Lace Bikini Panty                       4.4            91   
Wacoal Women's Slimline Seamless Minimizer Bra         4.3           298   
Hanky Panky Womens Signature Lace Retro V-Kini         4.4            46   
Wacoal Women's Red Carpet Strapless Bra                4.4           747   

                                                                                     style_attributes  \
product_na

In [15]:
data_ae_com.head()

mrp      price  \
product_name                                             
Aerie Everyday Loves Lace Cheeky  12.50 USD  12.50 USD   
Aerie Everyday Loves Lace Cheeky  12.50 USD  12.50 USD   
Aerie Everyday Loves Lace Cheeky  12.50 USD  12.50 USD   
Aerie Everyday Loves Lace Cheeky  12.50 USD  12.50 USD   
Aerie Everyday Loves Lace Cheeky  12.50 USD  12.50 USD   

                                                                            pdp_url  \
product_name                                                                          
Aerie Everyday Loves Lace Cheeky  https://www.ae.com/aerie-everyday-loves-lace-c...   
Aerie Everyday Loves Lace Cheeky  https://www.ae.com/aerie-everyday-loves-lace-c...   
Aerie Everyday Loves Lace Cheeky  https://www.ae.com/aerie-everyday-loves-lace-c...   
Aerie Everyday Loves Lace Cheeky  https://www.ae.com/aerie-everyday-loves-lace-c...   
Aerie Everyday Loves Lace Cheeky  https://www.ae.com/aerie-everyday-loves-lace-c...   

                                 brand_name product_category retailer  \
product_name                                                            
Aerie Everyday Loves Lace Cheeky      AERIE         Cheekies    Ae US   
Aerie Everyday Loves Lace Cheeky      AERIE         Cheekies    Ae US   
Aerie Everyday Loves Lace Cheeky      AERIE         Cheekies    Ae US   
Aerie Everyday Loves Lace Cheeky      AERIE         Cheekies    Ae US   
Aerie Everyday Loves Lace Cheeky      AERIE         Cheekies    Ae US   

                                                                        description  \
product_name                                                                          
Aerie Everyday Loves Lace Cheeky  Introducing Everyday Loves™: Made with love. E...   
Aerie Everyday Loves Lace Cheeky  Introducing Everyday Loves™: Made with love. E...   
Aerie Everyday Loves Lace Cheeky  Introducing Everyday Loves™: Made with love. E...   
Aerie Everyday Loves Lace Cheeky  Introducing Everyday Loves™: Made with love. E...   
Aerie Everyday Loves Lace Cheeky  Introducing Everyday Loves™: Made with love. E...   

                                  rating  review_count  \
product_name                                             
Aerie Everyday Loves Lace Cheeky     5.0           8.0   
Aerie Everyday Loves Lace Cheeky     5.0           8.0   
Aerie Everyday Loves Lace Cheeky     5.0           8.0   
Aerie Everyday Loves Lace Cheeky     5.0           8.0   
Aerie Everyday Loves Lace Cheeky     5.0           8.0   

                                                                   style_attributes  \
product_name                                                                          
Aerie Everyday Loves Lace Cheeky  ["Soft lace with the right amount of stretch",...   
Aerie Everyday Loves Lace Cheeky  ["Soft lace with the right amount of stretch",...   
Aerie Everyday Loves Lace Cheeky  ["Soft lace with the right amount of stretch",...   
Aerie Everyday Loves Lace Cheeky  ["Soft lace with the right amount of stretch",...   
Aerie Everyday Loves Lace Cheeky  ["Soft lace with the right amount of stretch",...   

                                                         total_sizes  \
product_name                                                           
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   

                                                      available_size  \
product_name                                                           
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Loves Lace Cheeky  ["XS", "S", "M", "L", "XL", "XXL"]   
Aerie Everyday Love

# analisis de cada dataframe

usamos el metodo .info para obtener información general de los dataframe

In [16]:
data_victoriassecret.info()

<class 'pandas.core.frame.DataFrame'>
Index: 453386 entries, Very Sexy Strappy Lace Thong Panty to Dream Angels NEW! Lace Mini Slip
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   mrp               453386 non-null  object 
 1   price             453386 non-null  object 
 2   pdp_url           453386 non-null  object 
 3   brand_name        453386 non-null  object 
 4   product_category  453386 non-null  object 
 5   retailer          453386 non-null  object 
 6   description       453386 non-null  object 
 7   rating            137734 non-null  float64
 8   review_count      137734 non-null  float64
 9   style_attributes  0 non-null       float64
 10  total_sizes       453386 non-null  object 
 11  available_size    453386 non-null  object 
 12  color             453386 non-null  object 
dtypes: float64(3), object(10)
memory usage: 48.4+ MB


In [17]:
data_us_topshop_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3082 entries, MATERNITY Criss Cross Knickers to Lace Trim Triangle Bra
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               3082 non-null   object 
 1   price             3082 non-null   object 
 2   pdp_url           3082 non-null   object 
 3   brand_name        3082 non-null   object 
 4   product_category  3082 non-null   object 
 5   retailer          3082 non-null   object 
 6   description       3082 non-null   object 
 7   rating            266 non-null    float64
 8   review_count      266 non-null    float64
 9   style_attributes  0 non-null      float64
 10  total_sizes       3082 non-null   object 
 11  available_size    3082 non-null   object 
 12  color             2405 non-null   object 
dtypes: float64(3), object(10)
memory usage: 337.1+ KB


In [18]:
data_shop_nordstrom_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 12568 entries, 'B Fitting' High Cut Briefs to Plunge Underwire Push-Up Bra
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               12568 non-null  object 
 1   price             12568 non-null  object 
 2   pdp_url           12568 non-null  object 
 3   brand_name        12568 non-null  object 
 4   product_category  12568 non-null  object 
 5   retailer          12568 non-null  object 
 6   description       12568 non-null  object 
 7   rating            7878 non-null   float64
 8   review_count      7878 non-null   float64
 9   style_attributes  12568 non-null  object 
 10  total_sizes       12568 non-null  object 
 11  available_size    12568 non-null  object 
 12  color             12568 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.3+ MB


In [19]:
data_macys_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 40897 entries, ID String Bikini QF1754 to Modern Logo Pants D1632
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               40897 non-null  object 
 1   price             40897 non-null  object 
 2   pdp_url           40897 non-null  object 
 3   brand_name        40897 non-null  object 
 4   product_category  40897 non-null  object 
 5   retailer          40897 non-null  object 
 6   description       40897 non-null  object 
 7   rating            26092 non-null  float64
 8   review_count      26101 non-null  float64
 9   style_attributes  40897 non-null  object 
 10  total_sizes       40897 non-null  object 
 11  available_size    40897 non-null  object 
 12  color             40897 non-null  object 
dtypes: float64(2), object(11)
memory usage: 4.4+ MB


In [20]:
data_hankypanky_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 35005 entries, Silky 20” A-line Half Slip with Lace to Retro Lace *Plus Size* V-kini
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               35005 non-null  object 
 1   price             35005 non-null  object 
 2   pdp_url           35005 non-null  object 
 3   brand_name        35005 non-null  object 
 4   product_category  35005 non-null  object 
 5   retailer          35005 non-null  object 
 6   description       35005 non-null  object 
 7   rating            8849 non-null   float64
 8   review_count      8849 non-null   float64
 9   style_attributes  35005 non-null  object 
 10  total_sizes       35005 non-null  object 
 11  available_size    35005 non-null  object 
 12  color             35005 non-null  object 
dtypes: float64(2), object(11)
memory usage: 3.7+ MB


In [21]:
data_calvinklein_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4747 entries, logo cotton stretch thong to thong with lace
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               4747 non-null   object 
 1   price             4747 non-null   object 
 2   pdp_url           4747 non-null   object 
 3   brand_name        4747 non-null   object 
 4   product_category  4747 non-null   object 
 5   retailer          4747 non-null   object 
 6   description       4747 non-null   object 
 7   rating            0 non-null      float64
 8   review_count      0 non-null      float64
 9   style_attributes  4747 non-null   object 
 10  total_sizes       4747 non-null   object 
 11  available_size    4747 non-null   object 
 12  color             4747 non-null   object 
dtypes: float64(2), object(11)
memory usage: 519.2+ KB


In [22]:
data_btemptd_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3518 entries, b.tempt'd Ciao Bella Bralette to b.splendid Bralette
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               3518 non-null   object 
 1   price             3518 non-null   object 
 2   pdp_url           3518 non-null   object 
 3   brand_name        3518 non-null   object 
 4   product_category  3518 non-null   object 
 5   retailer          3518 non-null   object 
 6   description       3518 non-null   object 
 7   rating            0 non-null      float64
 8   review_count      0 non-null      float64
 9   style_attributes  3518 non-null   object 
 10  total_sizes       3518 non-null   object 
 11  available_size    3518 non-null   object 
 12  color             3518 non-null   object 
dtypes: float64(2), object(11)
memory usage: 384.8+ KB


In [23]:
data_amazon_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 31612 entries, Calvin Klein Women's Sheer Marquisette Demi Unlined Bra to Wacoal Women's Retro Chic Underwire Bra
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               31612 non-null  object 
 1   price             31612 non-null  object 
 2   pdp_url           31612 non-null  object 
 3   brand_name        31612 non-null  object 
 4   product_category  31612 non-null  object 
 5   retailer          31612 non-null  object 
 6   description       31612 non-null  object 
 7   rating            31612 non-null  float64
 8   review_count      31612 non-null  int64  
 9   style_attributes  31612 non-null  object 
 10  total_sizes       31612 non-null  object 
 11  available_size    31612 non-null  object 
 12  color             31612 non-null  object 
dtypes: float64(1), int64(1), object(11)
memory usage: 3.4+ MB


In [24]:
data_ae_com.info()

<class 'pandas.core.frame.DataFrame'>
Index: 28328 entries, Aerie Everyday Loves Lace Cheeky to Aerie Hi-Neck Lace Trim Bralette
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   mrp               28328 non-null  object 
 1   price             28328 non-null  object 
 2   pdp_url           28328 non-null  object 
 3   brand_name        28328 non-null  object 
 4   product_category  28328 non-null  object 
 5   retailer          28328 non-null  object 
 6   description       28328 non-null  object 
 7   rating            21125 non-null  float64
 8   review_count      9590 non-null   float64
 9   style_attributes  28328 non-null  object 
 10  total_sizes       28328 non-null  object 
 11  available_size    28328 non-null  object 
 12  color             28328 non-null  object 
dtypes: float64(2), object(11)
memory usage: 3.0+ MB


# nota
* se observa que en 8 de los 9 dataframe las columnas rating y review_count, ademas hay dos dataframe que ademas tine la columna style_attributes junto con las otras, estos dataframe son data_victoriassecret y data_us_topshop_com
* el dataframe data_amazon_com, no muestra valores faltantes en estas inspeción inicial

In [94]:
data_victoriassecret.value_counts(["price"], sort=False)

price    
$10.50       45832
$12.95          29
$12.99        4917
$14.50        5411
$14.99       12701
$15.00        1851
$16.50        4166
$16.95          24
$16.99        5052
$17.99        2438
$19.99       22784
$2.99          136
$20.00       15435
$21.99         232
$22.50         132
$22.95        1025
$23.99         279
$24.00          29
$24.50        9800
$24.95        2413
$24.99        9029
$25.00        3391
$26.95         767
$27.95          57
$28.00         347
$28.95         120
$29.50       14140
$29.95         134
$29.99       11172
$3.99         2795
$30.00         926
$32.00         915
$32.50        3973
$32.95       16968
$32.99          12
$34.00          72
$34.50       24636
$34.95       24148
$34.99        3886
$35.99          42
$36.00        4326
$36.50       33365
$36.95        3300
$38.00        3293
$38.50        1420
$39.50       14006
$39.95         946
$39.99         133
$4.99         1002
$40.00         462
$42.00        1039
$42.50        8182
$4

In [91]:
data_us_topshop_com.value_counts(["price"], sort=False)

price                                                                                                                                                          
 $10.00                                                                                                                                                            905
 $10.00\n\t\t\t\t\t            \t\t\t\t\t\t\t\t            \t\t\t\t\t            \t\t\t\n\t\t\t\t\t            \t\t\n\t\t\t\t\t\t            \t\t\t\tNow $4.00       2
 $110.00                                                                                                                                                            19
 $12.00                                                                                                                                                             47
 $12.00\n\t\t\t\t\t            \t\t\t\t\t\t\t\t            \t\t\t\t\t            \t\t\t\n\t\t\t\t\t            \t\t\n\t\t\t\t\t\t            \t\t\t\tNow $4.00       2
 $15.

In [92]:
data_shop_nordstrom_com.value_counts(["price"], sort=False)

price        
$100.00            36
$102.00            11
$11.98             37
$13.00           1789
$13.20             83
$135.00            12
$14.00             40
$14.98             11
$15.00            669
$15.40            171
$16.00             11
$16.80             51
$17.40              2
$17.98             10
$18.00            378
$19.00            110
$19.20              9
$19.60             32
$20.00            978
$20.40             19
$20.98             43
$21.60             39
$22.00            523
$22.40             16
$22.80             29
$23.00            177
$23.00–$25.00     153
$23.40             59
$24.00            142
$25.00            265
$25.20             99
$26.00            160
$26.40             30
$26.60            102
$27.00             32
$27.60              9
$28.00            229
$29.00             58
$29.40             44
$30.00            241
$31.00             70
$31.20–$33.60      22
$32.00            380
$32.20             99
$33.00            

In [102]:
data_macys_com.value_counts(["price"], sort=False)

price   
$10.00       192
$10.12        10
$10.50       109
$10.68        19
$104.00       12
$108.00       83
$11.25        53
$11.50       636
$11.55       379
$12.00      1412
$12.37       260
$12.60         8
$12.99         1
$120.00        1
$125.00       26
$13.00      4213
$13.12         3
$13.50         1
$13.65        32
$135.00       26
$14.00       161
$14.25         7
$14.62        22
$14.70        96
$14.99        57
$15.00      1188
$15.40        15
$15.75        41
$15.99        10
$16.50        49
$16.80        21
$16.87        19
$17.32        56
$17.50        95
$18.00       726
$18.20        45
$18.37        28
$18.56        24
$18.90       209
$19.00       199
$19.50        53
$19.68        20
$19.95       203
$19.99        10
$20.00      2214
$20.25       147
$20.47        31
$21.00       144
$21.37       147
$21.93        18
$22.00      1684
$22.05       103
$22.13        48
$22.40       118
$22.50        36
$23.00       421
$23.10       246
$23.62        70
$23.8

In [103]:
data_hankypanky_com.value_counts(["price"], sort=False)

price  
$10          25
$100         53
$102         75
$105         25
$106         52
$108        229
$11          25
$110        113
$112         64
$114          2
$116          8
$118         49
$12          14
$125         39
$126         25
$128         50
$135        175
$14          61
$14.99        6
$140        113
$15         104
$150         41
$158         13
$16.99       29
$160         88
$165         25
$18         400
$186         24
$19          73
$20        5103
$20.99        6
$21          25
$22        1103
$22.99       25
$23        1025
$23.99       12
$24         564
$25        1831
$25.99       29
$26         125
$26.99        4
$27         100
$28         650
$29         278
$295         50
$30        1649
$300         56
$31         126
$31.99       11
$32        2969
$32.50       98
$33          21
$34        1375
$35         485
$35.99       12
$36         650
$37          94
$38          74
$39        1291
$40         248
$41         100
$42         596


In [107]:
data_calvinklein_com.value_counts(["price"], sort=False)

price          
$12.00               40
$12.99               40
$13.00             2282
$14.99               82
$16.99               30
$18.00               24
$19.99              257
$19.99 - $24.99      12
$20.00              235
$22.00              131
$24.00              191
$24.99              143
$26.00               22
$28.00              207
$29.99               72
$32.00               59
$34.00               15
$34.50                2
$34.99                9
$36.00              101
$38.00              139
$39.00               92
$39.50               10
$39.99               10
$4.99                15
$40.00               45
$42.00               56
$44.00               63
$46.00               87
$48.00               36
$49.00               14
$50.00               14
$58.00               25
$59.00                2
$6.99               132
$72.00                7
$9.99                46
dtype: int64

In [106]:
data_btemptd_com.value_counts(["price"], sort=False)

price  
$10.80      90
$11.40      41
$12.00       9
$12.60     144
$13.00     596
$13.20      22
$13.30      33
$14.00       7
$14.40      26
$15.00      10
$15.40      21
$16.80       7
$17.50      14
$18.00     138
$18.20       6
$19.00      50
$19.20      13
$19.80      49
$20.00      52
$21.00      67
$22.00      56
$22.40      24
$22.80      26
$23.10      10
$23.40      24
$23.80       8
$24.00     108
$24.50       4
$25.20      76
$26.00       6
$26.40      96
$26.60      34
$27.00      14
$28.00      20
$29.40      16
$30.00      71
$30.80      73
$31.50       6
$32.00      46
$32.20      10
$34.00      16
$35.00       8
$36.00       6
$38.00      60
$40.00      91
$42.00      96
$44.00     138
$45.00      30
$46.00       4
$7.80      523
$9.10      423
dtype: int64

In [105]:
data_amazon_com.value_counts(["price"], sort=False)

price  
$10.00       28
$10.14        1
$10.25        2
$10.27       18
$10.32        1
$10.35        5
$10.36        1
$10.38        1
$10.40       18
$10.42        1
$10.49        7
$10.50       29
$10.56        3
$10.64        1
$10.80        2
$10.83        1
$10.89        6
$10.99       11
$11.00      123
$11.20        1
$11.23        1
$11.36        4
$11.49        1
$11.55        2
$11.65        5
$11.70        1
$11.75        1
$11.80        4
$11.87        1
$11.90        1
$11.95       52
$11.99       12
$12.00      132
$12.03        2
$12.07        5
$12.32        1
$12.35        1
$12.39        1
$12.43        1
$12.49        5
$12.60        1
$12.66        2
$12.74        5
$12.85        1
$12.87        1
$12.89        1
$12.90        6
$12.96        1
$12.97       12
$12.98        8
$12.99       59
$13.00       30
$13.05        1
$13.08        1
$13.09        1
$13.36        1
$13.39       17
$13.44        7
$13.60        1
$13.69        3
$13.70       11
$13.87        1


In [104]:
data_ae_com.value_counts(["price"], sort=False)

price    
10.15 USD      23
10.85 USD      23
11.47 USD     118
12.47 USD     212
12.50 USD    5786
13.47 USD    1484
13.77 USD     132
13.98 USD       6
14.50 USD     871
14.97 USD    1070
16.06 USD     139
16.17 USD    1282
17.46 USD     242
17.47 USD     368
17.97 USD    1807
17.98 USD       2
18.86 USD     561
19.77 USD     104
19.97 USD     128
20.96 USD     915
20.97 USD     821
22.17 USD      30
22.47 USD       8
23.06 USD     248
23.97 USD     549
24.46 USD     542
24.97 USD     187
25.55 USD      46
25.86 USD      88
26.95 USD      46
26.97 USD     329
27.96 USD    1022
29.97 USD     500
3.99 USD       18
31.46 USD     705
34.96 USD     720
46.89 USD      35
48.99 USD      15
51.09 USD      56
7.50 USD      253
8.05 USD       24
9.50 USD     6813
dtype: int64

# ¿Cómo difiere el precio según la marca?
victoriassecret: los precios de esta marca varian entre $10.50 - $98.00 

topshop: los precios de esta marca varian entre $10.00 - $90.00 

shop nordstrom: los precios de esta marca varian entre $11.98 - $135.00

macys: los precios de esta marca varian entre $10.00 - $135.00

hankypanky: los precios de esta marca varian entre $10.00 - $300

calvinklein: los precios de esta marca varian entre $12.00 - $72.00 

btemptd: los precios de esta marca varian entre $10.80 - $46.00

amazon: los precios de esta marca varian entre $10.00 - $80.99 

ae: los precios de esta marca varian entre $10.15 - $51.09

NOTA: los valores mostrados estan es dolares Estado Unidenses

In [122]:
pd.set_option("max_colwidth", None)

In [124]:
data_victoriassecret["description"].head(1)

product_name
Very Sexy Strappy Lace Thong Panty    Lots of cheek peek, pretty lace, a strappy back—this sexy panty is so not subtle. Allover lace with front bow V-back with crisscross straps Low rise Minimal back coverage: lots of cheek peek Imported nylon/spandex
Name: description, dtype: object

In [127]:
data_victoriassecret.value_counts(["color"])

color                                              
Black                                                  17474
black                                                  13134
White                                                  10803
pure black                                              7101
Ensign                                                  6738
Almost Nude                                             6271
bayberry                                                5441
Hello Lovely                                            5159
Sheer Pink                                              4985
Radiating Aztec                                         4862
Black Marl                                              4740
buff                                                    4576
Burnished Lilac                                         4518
triumph white                                           4265
coconut white                                           3488
Black Lace                       

In [137]:
data_us_topshop_com.value_counts(["color"])

color         
BLACK             448
MULTI             179
WHITE             111
RED                95
IVORY              93
PINK               87
NAVY BLUE          80
Nude               74
AQUA               68
LILAC              64
CHARCOAL           61
PALE BLUE          60
GREY MARL          54
CHARTREUSE         50
CORAL              50
TEAL               44
PALE PINK          35
Pink               34
ROSE               34
Dark Nude          33
BURNT ORANGE       33
PLUM               33
RASPBERRY          32
Blush              31
Bright Coral       29
MAGENTA            28
BLUE               27
KHAKI              27
Flur Pink          23
Bubblegum          23
Ivory              22
YELLOW             22
Pale Pink          22
CREAM              21
LAVENDER           19
MAUVE              19
Bright Pink        17
Sugar Pink         16
Monochrome         14
COBALT             14
APPLE              14
INDIGO             13
Blue               12
pink               12
FUCHSIA          

In [136]:
data_shop_nordstrom_com.value_counts(["color"])

color                         
Black                             2551
White                              815
Bare                               396
Naturally Nude                     352
Chai                               333
Ivory                              302
Nude                               219
Light Ivory                        195
Sand                               186
Night                              163
Beige Soft                         130
Natural Nude                       121
Pink Yarrow                        117
Off White                          112
Blue Turquoise                     109
Campanula                          109
Marshmallow                        106
Navy                               105
Multi                               93
Red                                 89
Black Heather                       87
Cappuccino                          87
Fresh Taupe                         85
Grey Heather                        81
Desert Sunset                    

In [135]:
data_macys_com.value_counts(["color"])

color                                                                   
Black                                                                       6533
White                                                                       1884
Night                                                                       1063
Bare                                                                        1010
Ivory                                                                        937
Naturally Nude                                                               857
Red                                                                          593
Sand                                                                         591
Chai                                                                         520
Multi                                                                        386
Cappuccino                                                                   373
Desert Sunset                       

In [134]:
data_hankypanky_com.value_counts(["color"])

color                         
Black                             4712
White                             1814
Chai                              1374
Red                               1227
Marshmallow                        883
Light Ivory                        757
Granite                            686
Leopard Nouveau                    605
Bliss Pink                         586
White/Powder Blue                  542
Vanilla                            528
Coral Gables                       519
Taupe                              491
Ivory                              489
Tickled Pink                       481
Black/Heather                      386
Navy                               378
Mocha                              373
Storm Cloud Blue                   359
Capri Bloom                        348
Chambray                           346
Navy Blue                          290
Cygnet/Vanilla                     278
Sophisticat                        277
White/Baby Blue                  

In [133]:
data_calvinklein_com.value_counts(["color"])

color                              
BLACK                                  869
WHITE                                  426
BARE                                   242
POLYMER BLUE                           163
GREY                                   114
GREY HEATHER                           101
SALVIA                                  89
SALT LAKE                               85
CHARCOAL                                77
SULTRY                                  73
DESERT SUNSET                           70
UNDONE WITH SCANDAL LACE                63
NAVY                                    59
SCANDAL                                 57
EPHERMERAL                              55
BOUNDLESS LOGO                          54
FATHOM                                  53
STRIKING                                52
COLLAGED FLORAL WITH IVORY LACE         52
SENSATION                               50
NUDE                                    46
SPARROW                                 46
GRADIENT LOGO     

In [132]:
data_btemptd_com.value_counts(["color"])

color                           
Night                               444
Wild Aster                          167
Au Natural                          138
Bridal White                        118
Vanilla Ice                          92
Tango Red                            88
Marine                               82
Cappuccino                           78
Barrier Reef                         75
Dark Grey/Heather                    71
Au Natural/Heather                   71
Pansy Heather                        71
Limoges/Heather                      71
Pink Yarrow                          70
Placid Blue                          66
Blithe                               65
Pink Yarrow Heather                  63
Very Violet                          59
Pansy                                57
Peacoat Blue                         56
Baja Blue                            52
Wineberry                            52
Antler                               52
Lapis                                50
Blacken

In [131]:
data_amazon_com.value_counts(["color"])

color                                                         
Black                                                             5664
Naturally Nude                                                    2825
Ivory                                                             1778
White                                                             1480
Cappuccino                                                        1168
Toast                                                             1098
Bare                                                               954
Large                                                              857
Small                                                              845
Medium                                                             838
Almond Blossom                                                     442
Night                                                              384
Pecan                                                              369
Violet Tulip  

In [129]:
data_ae_com.value_counts(["color"])

color            
True Black           4401
Navy                 2184
Deep Plum            1720
White                1298
Natural Nude         1051
Dark Heather          875
Soft Muslin           854
Buff                  847
Current Coral         827
Alpha Turquoise       678
Creme Blue            666
Valentine             660
Slab Gray             646
Fresh Bright          636
Pink Tint             590
Fall Mint             554
Royal Navy            546
Stargaze              517
Maroon Lagoon         502
Vanity Peach          452
Coral                 441
Balmy Teal            389
Light Heather         381
Deep Burgundy         360
Heather Frost         348
Light Blue            344
Nude                  334
Darkness Falls        260
Ballet Pink           241
Red Cord              229
Stone                 216
Monaco Blue           205
Somber Navy           199
Olive Fun             187
Rugged Green          186
Berry                 165
Silver Shadow         163
Bright Blue         

# ¿Cuáles son los colores más comunes utilizados por las diferentes marcas?

se observa que los colores más usado en cada uno de las marcas es:

* victoriassecret: BLACK que es usado en 17474 productos

* topshop: BLACK que es usado en 448 productos

* shop nordstrom: Black que es usado en 2551 productos

* macys: Black que es usado en 6533 productos

* hankypanky: Black que es usado en 4712 productos

* calvinklein: BLACK que es usado en 869 productos

* btemptd: Night que es usado en 444 productos

* amazon: Black que es usado en 5664 productos 

* ae: True Black que es usado en 4401 productos

In [160]:
data_victoriassecret.value_counts(["review_count"], sort=False)

review_count
2.000000e+00    1265
3.000000e+00    1348
4.000000e+00    2625
5.000000e+00    1375
6.000000e+00    1433
7.000000e+00     719
8.000000e+00     938
9.000000e+00     951
1.000000e+01      85
1.100000e+01     288
1.200000e+01    2021
1.300000e+01     375
1.400000e+01      98
1.500000e+01    1007
1.600000e+01     648
1.700000e+01     418
1.800000e+01    1323
1.900000e+01     361
2.000000e+01     328
2.100000e+01     288
2.200000e+01     553
2.300000e+01     259
2.400000e+01     466
2.500000e+01    1319
2.600000e+01    1679
2.700000e+01     773
2.800000e+01     559
2.900000e+01    1574
3.000000e+01     909
3.100000e+01     119
3.200000e+01    1106
3.300000e+01    2224
3.400000e+01    1394
3.500000e+01    1501
3.600000e+01     683
3.700000e+01     340
3.800000e+01     909
3.900000e+01    1191
4.000000e+01     155
4.100000e+01    1246
4.200000e+01      69
4.300000e+01    1160
4.400000e+01     173
4.500000e+01     181
4.600000e+01      19
4.700000e+01    1100
4.800000e+01      97


In [161]:
data_ae_com.value_counts(["review_count"], sort=False)

review_count
1.0             2756
2.0             1824
3.0              921
4.0              625
5.0              837
6.0              275
7.0              154
8.0              231
9.0              128
10.0             353
11.0             209
12.0             108
13.0              55
14.0             186
15.0              46
18.0              96
26.0              10
29.0             272
32.0             168
63.0             120
92.0             216
dtype: int64

In [162]:
data_us_topshop_com.value_counts(["review_count"], sort=False)

review_count
1.0             165
2.0              49
3.0              20
4.0              13
5.0               1
15.0             11
17.0              7
dtype: int64

In [170]:
data_shop_nordstrom_com.value_counts(["review_count"])

review_count
1.0             635
2.0             496
3.0             464
95.0            415
94.0            349
8.0             269
5.0             267
4.0             261
7.0             186
10.0            170
33.0            163
34.0            158
126.0           155
109.0           153
12.0            150
101.0           147
15.0            145
28.0            144
80.0            141
49.0            138
44.0            122
21.0            115
16.0            107
65.0            107
19.0             98
6.0              96
41.0             96
113.0            92
81.0             90
124.0            90
14.0             87
30.0             84
39.0             74
319.0            72
29.0             70
76.0             68
432.0            64
60.0             62
779.0            60
26.0             60
149.0            57
286.0            57
24.0             55
48.0             52
174.0            52
583.0            51
135.0            51
171.0            48
20.0             48
203.0  

In [169]:
data_macys_com.value_counts(["review_count"])

review_count
1.0             4767
2.0             2685
3.0             1851
5.0             1752
7.0             1131
6.0              985
11.0             977
13.0             945
8.0              632
4.0              573
25.0             377
15.0             376
57.0             374
21.0             371
12.0             321
9.0              316
16.0             313
20.0             312
68.0             273
17.0             262
10.0             227
198.0            210
14.0             201
69.0             200
58.0             196
78.0             195
19.0             194
217.0            183
48.0             162
125.0            155
51.0             150
31.0             149
52.0             149
18.0             148
27.0             143
26.0             130
124.0            109
128.0            108
178.0            107
129.0            107
24.0             106
64.0             105
65.0             105
286.0            105
354.0            104
50.0             102
166.0             96


In [168]:
data_hankypanky_com.value_counts(["review_count"]) 

review_count
1.0             3142
7.0             1541
2.0             1290
4.0              927
8.0              679
3.0              600
10.0             405
6.0              265
dtype: int64

In [159]:
data_calvinklein_com.value_counts(["review_count"])
#la columna review_count se encuentra completamente vacia

Series([], dtype: int64)

In [152]:
data_btemptd_com.value_counts(["review_count"]) 
#la columna review_count se encuentra completame vacia

Series([], dtype: int64)

In [172]:
data_amazon_com.value_counts(["review_count"])

review_count
377             923
753             779
585             732
375             697
754             657
376             635
755             632
378             547
280             542
31              527
61              481
94              469
107             443
90              430
584             392
381             373
961             338
410             337
2               333
1137            319
12              309
57              303
382             296
467             291
247             283
30              276
466             275
194             273
601             268
1145            268
71              261
530             259
136             257
730             251
972             248
60              243
17              243
32              231
806             228
58              226
193             224
768             219
414             214
1143            207
765             203
766             200
46              199
5               196
380             192
308    

# Análisis de las calificaciones de los productos (cuando corresponda)

victoriassecret: 

topshop: 

shop nordstrom: 

macys: 

hankypanky: 

calvinklein: 

btemptd: 

amazon: 

ae: 